orbit/pkg/platform/platform_windows.go

```diff
@@ -341,3 +344,203 @@ func GetSMBiosUUID() (string, UUIDSource, error) {
+func getWorkingPath() (string, error) {
+	// getting current executable fullpath
+	exec, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	// returns the current executable directory
+	return filepath.Dir(exec), nil
+}
+
+// getOrbitVersion returns the version of the Orbit executable
+func getOrbitVersion(path string) (string, error) {
+	const (
+		expectedPrefix      = "orbit "
+		expectedVersionFlag = "-version"
+	)
+
+	if len(path) == 0 {
+		return "", errors.New("input executable is empty")
+	}
+
+	// running the executable with the version flag
+	args := []string{expectedVersionFlag}
+	out, err := exec.Command(path, args...).Output()
+	if err != nil {
+		return "", errors.New("there was a problem running target executable")
+	}
+
+	// parsing the output
+	versionOutputStr := string(out)
+	if len(versionOutputStr) == 0 {
+		return "", errors.New("empty executable output")
+	}
+
+	outputByLines := strings.Split(strings.TrimRight(versionOutputStr, "\n"), "\n")
+	if len(outputByLines) < 1 {
+		return "", errors.New("expected number of lines is not present")
+	}
+
+	rawVersionStr := strings.TrimSpace(strings.ToLower(outputByLines[0]))
+	if !strings.HasPrefix(rawVersionStr, expectedPrefix) {
+		return "", errors.New("expected version prefix is not present")
+	}
+
+	// getting the actual version string
+	versionStr := strings.ReplaceAll(rawVersionStr, expectedPrefix, "")
+	if len(versionStr) == 0 {
+		return "", errors.New("expected version information is not present")
+	}
+
+	return versionStr, nil
+}
+
+// fixSymlinkNotPresent fixes the issue where the symlink to the orbit service binary is not present
+// this is a workaround for the issue described here https://github.com/fleetdm/fleet/issues/10300
+func fixSymlinkNotPresent() error {
+	// getting current working directory
+	execPath, err := getWorkingPath()
+	if err != nil {
+		return err
+	}
+
+	// getting the path to orbit service binary
+	orbitPath := execPath + "\\..\\bin\\orbit\\orbit.exe"
+
+	// gathering target orbit version
+	versionOrbit, err := getOrbitVersion(orbitPath)
+	if err != nil {
+		return fmt.Errorf("getting orbit version: %w", err)
+	}
+
+	// checking if target orbit has the problematic logic
+	if versionOrbit != "1.6.0" && versionOrbit != "1.7.0" {
+		return nil
+	}
+
+	// checking if the orbit service binary symlink needs to be regenerated
+	_, err = os.Readlink(orbitPath)
+
+	// if there are no errors, it means that the symlink is present, nothing to do
+	if err == nil {
+		return nil
+	}
+
+	// handling error by renaming the locked binary file, marking it for deletion on reboot and
+	// regenerating the symlink
+	if !errors.Is(err, os.ErrNotExist) {
+
+		// symlink is going to be recreated - so better check that symlink target is present
+		targetSymlinkPath := ""
+		stableOrbitPath := execPath + "\\..\\bin\\orbit\\windows\\stable\\orbit.exe"
+		edgeOrbitPath := execPath + "\\..\\bin\\orbit\\windows\\edge\\orbit.exe"
```

Comment:
Users can use versions as channels too, e.g.:
```sh
fleetctl package [...] --orbit-channel 1.6
fleetctl package [...] --orbit-channel 1
fleetctl package [...] --orbit-channel 1.6.0
```
and that would end up with different directories (`opt/orbit/bin/orbit/macos/1.6/orbit`, `opt/orbit/bin/orbit/macos/1/orbit`, etc.).

Reply:
How about we diff the one `...\bin\orbit.exe` with any of the existing channels (users might have changed from one channel to another at runtime?) and then pick the one that is exactly the same binary as target.

Reply:
Good idea, I'll implement logic to tackle this

Reply:
I had some back and forth with this and ended up implementing a way more straightforward logic that does not require dealing with update channels.